src/scripts/main.js

```diff
@@ -1 +1,90 @@
 'use strict';
+
+function showNotification(posTop, posRight, description, type) {
+  const notification = document.createElement('div');
+
+  notification.dataset.qa = 'notification';
+
+  notification.style.position = 'fixed';
+
+  notification.style.margin = '0';
+  notification.style.width = '300px';
+  notification.style.padding = '10px';
+  notification.style.borderRadius = '10px';
+
+  notification.classList.add(type);
+  notification.textContent = description;
+
+  notification.style.top = `${posTop}px`;
+  notification.style.right = `${posRight}px`;
+
+  document.body.append(notification);
+
+  setTimeout(() => (notification.style.display = 'none'), 2000);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeout = setTimeout(reject, 3_000, 'First promise was rejected');
+
+  document.addEventListener(
+    'click',
+    (e) => {
+      e.preventDefault();
+
+      clearTimeout(timeout);
+      resolve('First promise was resolved');
+    },
+    { once: true },
+  );
+});
+
+firstPromise
+  .then((message) => {
+    showNotification(10, 10, message, 'success');
+  })
+  .catch((message) => {
+    showNotification(10, 10, message, 'error');
+  });
+
+const secondPromise = new Promise((resolve) => {
+  const callback = (e) => {
+    e.preventDefault();
+
+    document.removeEventListener('click', callback);
+    document.removeEventListener('contextmenu', callback);
```

Comment:
In the second promise, the callback is removed from both 'click' and 'contextmenu' events after the first interaction. This is correct if you want to resolve the promise on the first interaction of either event. If you intend to handle multiple interactions, consider adjusting the logic.